frontend/simple/js/state.js

```diff
@@ -99,19 +117,45 @@ const getters = {
   unreadMessageCount (state) {
     let mailboxContract = store.getters.mailboxContract
     return mailboxContract && mailboxContract.messages.filter((msg) => !msg.read).length
+  },
+  // Logged In user's identity contract
+  identity (state) {
+    return state.users[state.loggedIn]
+  },
+  // Logged In user's picture
+  picture (state) {
+    let identity = store.getters.identity
+    return identity && state[identity] && state[identity].attributes.picture
+  },
+  // list of group names and contractIds
+  groupsByName (state) {
+    let groups = []
+    for (let [key, value] of Object.entries(state.contracts)) {
+      if (value === 'GroupContract') {
+        groups.push({groupName: state[key].groupName, contractId: key})
+      }
+    }
+    return groups
+  },
+  // return a picture from a known user
+  getUserPicture (state) {
+    return function (username) {
+      return state.users[ username ] && state[ state.users[ username ] ] && state[ state.users[ username ] ].attributes.picture
+    }
   }
 }
 
 const actions = {
   // Used to update contracts to the current state that the server is aware of
-  async syncContractWithServer ({dispatch}: {dispatch: Function}, contractId: string) {
+  async syncContractWithServer ({dispatch, state}: {dispatch: Function, state: Object}, contractId: string) {
     let latest = await backend.latestHash(contractId)
-    let recent = await db.recentHash(contractId)
-    if (latest !== recent) {
+    // there is a chance two users are logged in to the same machine and must check their contracts before syncing
+    let recent = state.contracts[contractId] ? await db.recentHash(contractId) : null
+    if (latest !== recent || !state.contracts[contractId]) {
       // TODO Do we need a since call that is inclusive? Since does not imply inclusion
       let events = await backend.eventsSince(contractId, recent || contractId)
       // remove the first element in cases where we are not getting the contract for the first time
-      recent && events.shift()
+      state.contracts[contractId] && events.shift()
```

Comment:
Because it is a better determinant of whether this is the user's first time syncing this contract. As mentioned before the db doesn't store ownership and the recent comes from the db